examples/trusted_information_retrieval/module_0/rust/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Private Information Retrieval example for Project Oak.
+//!
+//! This shows how an Oak Node can retrieve information from an external database and process
+//! private queries from clients.
+//!
+//! Current example implementation uses a list of Santander Cycles in London as test database:
+//! https://tfl.gov.uk/tfl/syndication/feeds/cycle-hire/livecyclehireupdates.xml
+//!
+//! Clients send their location coordinates (latitude and longitude) and the Oak Node returns the
+//! location of the closest Point Of Interest (in our case a bike station).
+
+pub mod config;
+pub mod proto {
+    include!(concat!(
+        env!("OUT_DIR"),
+        "/oak.examples.trusted_information_retrieval.rs",
+    ));
+}
+#[cfg(test)]
+mod tests;
+
+use config::get_database_url;
+use database_proxy::get_database_entry;
+use log::{debug, error};
+use oak::grpc;
+use prost::Message;
+use proto::{
+    GetPointOfInterestRequest, GetPointOfInterestResponse, TrustedInformationRetrieval,
+    TrustedInformationRetrievalDispatcher,
+};
+
+/// Oak Node that connects to the database proxy Node.
+pub struct TrustedInformationRetrievalNode {
+    database_url: String,
+}
+
+/// A gRPC service implementation for the Private Information Retrieval example.
+impl TrustedInformationRetrieval for TrustedInformationRetrievalNode {
+    /// Get a point of interest based on requested id.
+    fn get_point_of_interest(
+        &mut self,
+        request: GetPointOfInterestRequest,
+    ) -> grpc::Result<GetPointOfInterestResponse> {
+        debug!("Received request: {:?}", request);
+
+        // Get requested database entry from the database proxy Node.
+        let entry = get_database_entry(&request.id, &self.database_url).map_err(|error| {
+            let status = grpc::build_status(grpc::Code::Unavailable, "");
+            error!("Couldn't get point of interest: {:?}, {:?}", error, status);
+            status
```

Comment:
Why Unavailable? Is the intention that the client should just retry? Maybe add some comments.